apiserver/common/controllerconfig.go

```diff
@@ -76,15 +77,44 @@ func (s *controllerStateShim) ControllerInfo(modelUUID string) (addrs []string,
 		return StateControllerInfo(s.State)
 	}
 
-	// Now check any external controllers.
 	ec := state.NewExternalControllers(s.State)
 	info, err := ec.ControllerForModel(modelUUID)
+
+	// the model may have been migrated, if that is the case we try to get the information.
+	// If that is true we try to save the migration controller as an external controller.
+	if errors.IsNotFound(err) {
+		migration, err := s.State.LatestRemovedModelMigrationByUUID(modelUUID)
+		if err != nil {
+			return handleControllerInfoErr(err)
```

Comment:
we might want to add information here. Something along the line of:
`The model might be migrated from an external controller to begin with, therefore we cannot know whether this model has been migrated or not in general...`